llvm/lib/Target/NVPTX/NVPTXIntrinsics.td

```diff
@@ -1520,11 +1520,13 @@ multiclass F_ATOMIC_2_imp<ValueType ptrT, NVPTXRegClass ptrclass,
   def imm : NVPTXInst<(outs regclass:$dst), (ins ptrclass:$addr, IMMType:$b),
     !strconcat("atom", SpaceStr, OpcStr, TypeStr, " \t$dst, [$addr], $b;", ""),
     [(set (regT regclass:$dst), (IntOp (ptrT ptrclass:$addr), IMM:$b))]>,
-  Requires<Pred>;
+  Requires<!listconcat(Pred, [!if(!eq(TypeStr, ".f16"), Predicate<"false">, Predicate<"true">)])>;
```

Comment:
This is a bit ugly, maybe you have an idea how to improve this? At least it seems to do what it should :)

Reply:
Considering that we want to disable `imm` completely, a somewhat simpler way would be to do
```
Requires<!if(!eq(TypeStr, ".f16"), [Predicate<"false">], Preds)])
```

Reply:
Thanks for the suggestion, this looks much nicer :)
Done.